File: test/leaps.jl
using DataFrames

## generate data (and export so R can read)

srand(1)
X = rand(100, 6)
y = rand(100)

d = convert(DataFrame, X)
d[:y] = y


writetable(joinpath(Pkg.dir("Leaps"), "test", "test.csv"), d)

using Leaps

## test normal case
rs = RegSubsets(X, y, intercept=false)
xhaust!(rs)

## test case with extra linear dependency
rs2 = RegSubsets(hcat(X[:,1], X), y, intercept=false)
xhaust!(rs2)

File: test/leaps.R
d <- data.frame(lapply(read.csv('test.csv', sep=',', dec='.', stringsAsFactors=F),
                       as.numeric))

library(leaps)

x1 <- as.matrix(d[, 1:6])
y <- d$y

x2 <- as.matrix(d[, c(1, 1:6)])

x3 <- as.matrix(d[, c(1:6, 1)])


x <- x1
wt=rep(1,length(y))
force.in=NULL


force.out=NULL
intercept=FALSE
nvmax=8
nbest=1
warn.dep=TRUE
